spice/training/actions.py

```diff
@@ -603,7 +679,19 @@ def train_model(self):
         train_dataset = self._load_dataset(config, "train")
 
         # tokenize dataset
-        tokenized_dataset = self._tokenize_dataset(train_dataset, config, "train")
+        image_processor = None
+        if base_model_repo_id == "spicecloud/spice-cnn-base":
+            image_processor, tokenized_dataset = self._process_dataset(
+                train_dataset, config, "train"
+            )
+        elif base_model_repo_id == "bert-base-uncased":
+            tokenized_dataset = self._tokenize_dataset(train_dataset, config, "train")
+        else:
+            error_message = (
+                f"train_model has unknown base model: {base_model_repo_id} type"
+            )
+            LOGGER.error(error_message)
+            raise ValueError(error_message)
```

Comment:
we'll definitely need to to change this to be based on some attribute of the training round like "model type" of "image" or "text", etc